csrc/optimization/optimize_consecutive_cast.cpp

```diff
@@ -0,0 +1,56 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <optimization/opt_pass.h>
+#include <ir_utils.h>
+
+namespace nvfuser::optimization {
+
+namespace {
+
+class ConsecutiveCastPass : OptimizationPass {
```

Comment:
This is an example of what a custom pass should be like.

We might have to add a header for this so we can actually run unit test in cpp.